### ISSUE TITLE
skylark_git_repository_test fails under standalone

### ISSUE THREAD

**Issue by werkt** [bazelbuild/bazel#4405]
### Description of the problem / feature request:

Bazel's `//src/test/shell/bazel:skylark_git_repository_test` relies on execution under `test --spawn_strategy=sandboxed`.  Running it with `test --spawn_strategy=standalone` causes the test to fail with errors during `git_repository`, failing to locate the revision for reset. This appears to be less of a correctness issue for the test, than a minor environmental difference that isn't reflected in standalone execution, but should probably work without sandboxing on.

Caveat for swapping between these two options: test results will be cached, and not invalidated based on the switch.  Prefer test runs with `--nocache_test_results`.

This failure is also unreproducible without invoking the *test* itself with `--spawn_strategy=standalone`; if only the offending `bazel run` invocations are put under `standalone`, the tests do not fail.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

`bazel test --spawn_strategy=sandboxed //src/test/shell/bazel:skylark_git_repository_test`

### What operating system are you running Bazel on?

linux

### What's the output of `bazel info release`?

0.9.0

###  Have you found anything relevant by searching the web?

No

### Any other information, logs, or outputs that you want to share?

Log files from execution of 0.9.0 under standalone strategy are attached: [skylark_git_repository.log](https://github.com/bazelbuild/bazel/files/1613723/skylark_git_repository.log)

**Comment by katre** [comment]
My apologies for not looking into this sooner.

I have tried to reproduce this and I cannot. Here is my command:
```
$ bazel test --runs_per_test=10 --spawn_strategy=standalone --nocache_test_results  //src/test/shell/bazel:skylark_git_repository_test
INFO: Analysed target //src/test/shell/bazel:skylark_git_repository_test (0 packages loaded).
INFO: Found 1 test target...
Target //src/test/shell/bazel:skylark_git_repository_test up-to-date:
  bazel-bin/src/test/shell/bazel/skylark_git_repository_test
INFO: Elapsed time: 240.568s, Critical Path: 239.64s
INFO: Build completed successfully, 11 total actions

Executed 1 out of 1 test: 1 test passes.
```

Can you give more details of your bazelrc or gitconfig that might be causing problems?

**Comment by werkt** [comment]
No bazelrc specified or present
Stock 0.9.0 deb package (from the bazel.build install page) master bazelrc present
git 1.9.1 (ubuntu 14.04 LTS)
Reproduces with an absent .gitconfig
After an expunge, using the bazel 0.9.0 release on the 0.9.0 rev in git:

```
$ bazel info release
release 0.9.0
$ git rev-parse HEAD
f4d58293c69e9359141576d2da88665dc63f2467
$ git log --oneline | head -1
f4d5829 Release 0.9.0 (2017-12-19)
$ bazel clean --expunge
INFO: Starting clean (this may take a while). Consider using --async if the clean takes more than several minutes.
$ bazel test --spawn_strategy=standalone //src/test/shell/bazel:skylark_git_repository_test                                                                                                                      ........
INFO: Analysed target //src/test/shell/bazel:skylark_git_repository_test (205 packages loaded).
INFO: Found 1 test target...
INFO: From Generating Java (Immutable) proto_library @googleapis//:google_devtools_build_v1_build_events_proto:
google/devtools/build/v1/build_events.proto: warning: Import google/rpc/status.proto but not used.
INFO: From Generating Java (Immutable) proto_library @googleapis//:google_bytestream_bytestream_proto:
google/bytestream/bytestream.proto: warning: Import google/protobuf/wrappers.proto but not used.
INFO: From Generating Java (Immutable) proto_library @googleapis//:google_watch_v1_proto:
google/watcher/v1/watch.proto: warning: Import google/protobuf/empty.proto but not used.
INFO: From Compiling src/main/cpp/blaze_util_posix.cc:
src/main/cpp/blaze_util_posix.cc: In function 'int blaze::ExecuteDaemon(const string&, const std::vector<std::basic_string<char> >&, const string&, const string&, blaze::BlazeServerStartup**)':
src/main/cpp/blaze_util_posix.cc:402:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
INFO: From Executing genrule //src/java_tools/junitrunner/java/com/google/testing/coverage:Jacoco_jarjar:
Jan 22, 2018 11:13:19 AM com.tonicsystems.jarjar.transform.JarTransformer transform
INFO: Transforming archive ZipArchive(/home/ggensure/.cache/bazel/_bazel_ggensure/1b47758cbd8b3cf858ef6c9ff0d39d05/execroot/io_bazel/bazel-out/k8-fastbuild/bin/src/java_tools/junitrunner/java/com/google/testing/coverage/JacocoCoverage_deploy.jar)
Jan 22, 2018 11:13:21 AM com.tonicsystems.jarjar.transform.jar.ClassTransformerJarProcessor process
WARNING: Failed to read class module-info.class: java.lang.RuntimeException: RemappingClassAdapter is deprecated, use ClassRemapper instead
INFO: From SkylarkAction external/googleapis/google_bytestream_bytestream_java_grpc_srcs.jar:
google/bytestream/bytestream.proto: warning: Import google/protobuf/wrappers.proto but not used.
INFO: From SkylarkAction external/googleapis/google_watch_v1_java_grpc_srcs.jar:
google/watcher/v1/watch.proto: warning: Import google/protobuf/empty.proto but not used.
FAIL: //src/test/shell/bazel:skylark_git_repository_test (see /home/ggensure/.cache/bazel/_bazel_ggensure/1b47758cbd8b3cf858ef6c9ff0d39d05/execroot/io_bazel/bazel-out/k8-fastbuild/testlogs/src/test/shell/bazel/skylark_git_repository_test/test.log)
Target //src/test/shell/bazel:skylark_git_repository_test up-to-date:
  bazel-bin/src/test/shell/bazel/skylark_git_repository_test
INFO: Elapsed time: 228.733s, Critical Path: 69.05s
INFO: Build completed, 1 test FAILED, 1883 total actions
//src/test/shell/bazel:skylark_git_repository_test                       FAILED in 42.5s
  /home/ggensure/.cache/bazel/_bazel_ggensure/1b47758cbd8b3cf858ef6c9ff0d39d05/execroot/io_bazel/bazel-out/k8-fastbuild/testlogs/src/test/shell/bazel/skylark_git_repository_test/test.log

Executed 1 out of 1 test: 1 fails locally.
```

All shell invoked `bazel` there is the 0.9.0 from deb package.  Problem reproduces equivalently with bazel copied and invoked as produced into `bazel-bin/src/bazel`. Test log is [attached](https://github.com/bazelbuild/bazel/files/1652681/test.log) (and looks similar to original).

**Comment by katre** [comment]
Okay, I couldn't reproduce this because it was fixed in 0ba8307ff46746864e357d0b7834b3c4247421d0, which should be in the 0.10.0 release.

Closing this now, thanks for reporting it!
